security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -29,15 +29,23 @@ import (
 	"github.com/gogo/protobuf/types"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
 	"google.golang.org/grpc/peer"
 	"google.golang.org/grpc/status"
 
 	"istio.io/istio/pilot/pkg/model"
 	"istio.io/istio/pkg/log"
 )
 
-// SecretType is used for secret discovery service to construct response.
-const SecretType = "type.googleapis.com/envoy.api.v2.Secret"
+const (
+	// SecretType is used for secret discovery service to construct response.
+	SecretType = "type.googleapis.com/envoy.api.v2.Secret"
+
+	// credentialTokenHeaderKey is the header key in gPRC header which is used to
+	// pass credential token from envoy to SDS.
+	// TODO(quanlin): update value after confirming what headerKey that client side uses.
```

Comment:
it should be set at `authorization` header.

Reply:
thanks(https://github.com/grpc/grpc/blob/master/src/core/lib/security/credentials/oauth2/oauth2_credentials.cc#L190 ?). /cc @htuch

another question, since sds server side needs to add checking token expire logic, can I assume that token passed from client is always in jwt format ?

Reply:
Yes that is the pointer.

I'm not quite sure whether it is always in JWT, it could be opaque oauth2 token. @qiwzhang?
For validating on server, see:
https://github.com/grpc/grpc-go/blob/master/Documentation/grpc-auth-support.md#validating-a-token-on-the-server